lib/amazon_selling_partners/client.rb

```diff
@@ -46,6 +47,42 @@ def initialize(sandbox: false, region: 'na', client_id: nil, client_secret: nil,
       @save_access_token = save_access_token
     end
 
+    def generate_authorization_url(
+      application_id:, # Our application id, looks like amzn1.sellerapps.app.0bf296b5-36a6-4942-a13e-random
+      state:, # A unique, short-lived value that is associated with our user.
+      redirect_uri: nil, # Optional, defaults to the first redirect url you've registered on your app
+      seller_central_url: 'https://sellercentral.amazon.com', # https://developer-docs.amazon.com/sp-api/docs/seller-central-urls,
+      draft: false # Set to true if our application is still in draft mode
+    )
+
+      url = "#{seller_central_url}/apps/authorize/consent?application_id=#{application_id}&state=#{state}"
+      url = "#{url}&redirect_uri=#{redirect_uri}" if redirect_uri.present?
+      url = "#{url}&version=beta" if draft
```

Comment:
If we would be loading rails here (ActiveSupport) we could do something like
```ruby
irb(main):002:0> draft = true
=> true
irb(main):003:0> {application_id: 'my_app_id', state: 'my_state', redirect_uri: 'https://my.url.com', version: (draft ? 'beta' : nil)}.to_query
=> "application_id=my_app_id&redirect_uri=https%3A%2F%2Fmy.url.com&state=my_state&version=beta"
```
But we are not. Luckily we can still use `URI.encode_www_form` to do that

```ruby
irb(main):005:0> URI.encode_www_form({application_id: 'my_app_id', state: 'my_state', redirect_uri: 'https://my.url.com', version: (draft ? 'beta' : nil)})
=> "application_id=my_app_id&state=my_state&redirect_uri=https%3A%2F%2Fmy.url.com&version=beta"
```

Reply:
@jozefvaclavik `{}.compact` has been part of ruby language since `v2.4.0`. We can use it 🙂